articles/building-an-appointment-scheduling-telegram-bot/index.md

```diff
@@ -0,0 +1,514 @@
+
+        context.bot.send_message(chat_id=chat_id, text="Welcome to Fauna Appointment Scheduler Bot \n\n To schedule an appointment enter /add_appointment \n To list al appointment enter /list_appointments \n To list all appoint,emts you have today enter /list_today_appointments")
+
+    except:
+
+        user = client.query(q.create(q.collection("Users"), {
+
+            "data": {
+
+                "id": chat_id,
+
+                "first_name": first_name,
+
+                "username": username,
+
+                "last_command": "",
+
+                "date": datetime.now(pytz.UTC)
+
+            }
+
+        }))
+
+        context.bot.send_message(chat_id=chat_id, text="Welcome to Fauna Appointment Scheduler Bot, your details have been saved 😊 \n\n To schedule an appointment enter /add_appointment \n To list al appointment enter /list_appointments \n To list all appointments you have today enter /list_today_appointments")
+
+
+```
+
+In the code above, you created a function called `start` then passed `update` and `context` as parameters. `update` is the telegram user’s data that is automatically passed from the dispatch handler. `context` is the bot instance passed from the dispatch handler. 
+
+You then collected the `chat_id`, `first_name`, and `username` from the updater then made a request to the FQL (Fauna Query Language) client using the `users_index` to check if the user with the `chat_id` already exists in the database. If the user exists, you send a welcome message. Else,  you create a user with the information provided.
+
+Next, you need to create a dispatch handler for the `/start` command. A command in telegram is any text that starts with `/`.
+
+You can also add other handlers such as image handler, text handler, regex handler, and many more. We are starting with a command handler because that is the first thing every user enters into a Telegram Bot.
+
+Copy and paste the code below at the end of your python file to call the dispatch handler to connect a command handler to the “start” command. The “start” command triggers the `start` method. The updater checks for messages from the user.
+
+```python
+
+dispatcher.add_handler(CommandHandler("start", start))
+
+updater.start_polling()
+
+```
+
+After updating your python file, run with the code below:
+
+
+```bash
+
+# python file name is app.py
+
+python app.py
+
+```
+While your python app is running, open your telegram app and search for the bot you created.
+
+![search_bot](/engineering-education/building-an-appointment-scheduling-telegram-bot/search_bot.png)
+
+Open your bot and start it by typing in the `/start` command. You will now receive a welcome message from the bot, as seen in the image below:
+
+![start_bot](/engineering-education/building-an-appointment-scheduling-telegram-bot/start_bot.png)
+
+#### Creating New Appointments
+
+To enable your telegram bot to create new appointments in the fauna database, copy and paste the code below in your python file.
```

Comment:
Now we will enable your telegram bot to create new appointments in the fauna database.